android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -324,9 +326,9 @@ private void updateNotifications() {
     if (!mCanUpdateNotifications) {
       return;
     }
-
+    Set<String> temp = new HashSet<>();
     List<PwPair> pwPairs = mPwCollection.getGroupedPwPairsSortedByRank(
-        Utils.newDistanceComparator());
+        Utils.newDistanceComparator(temp));
```

Comment:
Don't bother assigning to a local variable, just pass it directly, eg:
`Utils.newDistanceComparator(new HashSet<>())`

Alternatively, leave Utils.newDistanceComparator() as-is and create a new comparator that respects favorites.